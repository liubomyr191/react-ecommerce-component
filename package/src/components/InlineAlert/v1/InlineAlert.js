import React, { Component } from "react";
import PropTypes from "prop-types";
import styled, { css } from "styled-components";
import { withComponents } from "@reactioncommerce/components-context";
import { applyTheme, addTypographyStyles } from "../../../utils";

const StyledDiv = styled.div`
  ${addTypographyStyles("InlineAlert", "bodyText")};
  border-radius: ${applyTheme("InlineAlert.borderRadius")};
  border-style: ${applyTheme("InlineAlert.borderStyle")};
  border-width: ${applyTheme("InlineAlert.borderWidth")};
  padding-bottom: ${applyTheme("InlineAlert.paddingBottom")};
  padding-left: ${applyTheme("InlineAlert.paddingLeft")};
  padding-right: ${applyTheme("InlineAlert.paddingRight")};
  padding-top: ${applyTheme("InlineAlert.paddingTop")};
  position: relative;
  white-space: pre-wrap;
  overflow: hidden;
  display: ${({ isClosed }) => (isClosed ? "none" : "block")};
  opacity: ${({ isClosed }) => (isClosed ? 0 : 1)};
  transition: ${({ isClosed }) => (isClosed ? "opacity .25s ease-out, display 0s ease .25s" : "none")};
  ${(props) => {
    const { alertType } = props;
    switch (alertType) {
      case "error":
        return css`
          color: ${applyTheme("InlineAlert.color_error")};
          background-color: ${applyTheme("InlineAlert.backgroundColor_error")};
          border-color: ${applyTheme("InlineAlert.borderColor_error")};
        `;
      case "information":
        return css`
          color: ${applyTheme("InlineAlert.color_information")};
          background-color: ${applyTheme("InlineAlert.backgroundColor_information")};
          border-color: ${applyTheme("InlineAlert.borderColor_information")};
        `;
      case "success":
        return css`
          color: ${applyTheme("InlineAlert.color_success")};
          background-color: ${applyTheme("InlineAlert.backgroundColor_success")};
          border-color: ${applyTheme("InlineAlert.borderColor_success")};
        `;
      case "warning":
        return css`
          color: ${applyTheme("InlineAlert.color_warning")};
          background-color: ${applyTheme("InlineAlert.backgroundColor_warning")};
          border-color: ${applyTheme("InlineAlert.borderColor_warning")};
        `;
      default:
        return "";
    }
  }} `;

const StyledTitle = styled.div`
  ${addTypographyStyles("InlineAlert", "bodyTextSemiBold")};
  padding-bottom: 10px;
  `;

const StyledDismissButton = styled.button`
  background-color: transparent;
  border: none;
  cursor: pointer;
  height: 16px;
  padding: 0;
  position: absolute;
  right: 15px;
  top: 15px;
  width: 16px;
  `;

class InlineAlert extends Component {
  static propTypes = {
    /**
     * The type of alert
     */
    alertType: PropTypes.oneOf(["error", "information", "success", "warning"]).isRequired,
    /**
     * You can provide a `className` prop that will be applied to the outermost DOM element
     * rendered by this component. We do not recommend using this for styling purposes, but
     * it can be useful as a selector in some situations.
     */
    className: PropTypes.string,
    /**
     * If you've set up a components context using
     * [@reactioncommerce/components-context](https://github.com/reactioncommerce/components-context)
     * (recommended), then this prop will come from there automatically. If you have not
     * set up a components context or you want to override one of the components in a
     * single spot, you can pass in the components prop directly.
     */
    components: PropTypes.shape({
      /**
       * Pass an element (e.g., rendered SVG) to use as the Dismiss/Close icon
       */
      iconDismiss: PropTypes.node
    }).isRequired,
    /**
     * isAutoClosing: Closes automatically in 10 seconds
     */
    isAutoClosing: PropTypes.bool,
    /**
     * isClosed: Whether the alert is closed or open
     */
    isClosed: PropTypes.bool,
    /**
     * isDismissable: Display a Close/Dismiss button
     */
    isDismissable: PropTypes.bool,
    /**
     * Alert message
     */
    message: PropTypes.string.isRequired,
    /**
     * Alert title, optional
     */
    title: PropTypes.string
  };

  static defaultProps = {
    isAutoClosing: false,
    isClosed: false,
    isDismissable: false
  };

  state = {
    isClosed: this.props.isClosed
  };

  componentDidMount() {
    if (this.props.isAutoClosing === true) {
      setTimeout(() => {
        this.setState({ isClosed: true });
      }, 10000);
    }
  }

  handleDismissClick = (event) => {
    event.preventDefault();
    this.setState({ isClosed: true });
  };

  handleDismissKeyPress = (event) => {
    if (event.keyCode === 13) this.handleDismissClick(this.setState({ isClosed: true }));
  };

  autoClose = () => {
    window.setTimeout(10000, this.setState({ isClosed: true }));
  };

  render() {
    const { alertType, className, components, isDismissable, message, title } = this.props;
    const { iconDismiss } = components || {};
    const { isClosed } = this.state;

    return (
      <StyledDiv className={className} alertType={alertType} isClosed={isClosed}>
        {title
          ?
          <StyledTitle>{title}</StyledTitle>
          : null
        }
        {isDismissable
          ?
          <StyledDismissButton
            type="button"
            aria-label="close"
            onClick={this.handleDismissClick}
            onKeyPress={this.handleDismissKeyPress}
          >{iconDismiss}
          </StyledDismissButton>
          : null
        }
        {message}
      </StyledDiv>
    );
  }
}
const WrappedInlineAlert = withComponents(InlineAlert);

export default WrappedInlineAlert;
