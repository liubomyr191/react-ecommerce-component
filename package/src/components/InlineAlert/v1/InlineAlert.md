### Overview

Inline alerts are used to give action-based feedback messages and convey critical or informational account-related messages. Use inline alerts when a user needs more detailed information for an action.

```jsx noeditor
<InlineAlert/>
```

### Usage

Inline alerts are most often used when the user has taken an action. Alert messages appear in context and communicate when that action is successful, unsuccessful, or that it otherwise needs attention and further context.

Alert language should be polite, clear and concise. Alert language can also include a link. Optionally, a title can be added to an inline alert to give clarity, or when there are 2 or more lines of information to display.

Inline alerts should guide the user into taking corrective action if necessary. If the alert relates to a form, form field validation should be used in conjunction with the alert.

User should be able to dismiss inline alerts when appropriate. Information and success alerts can close automaticallly after 10 seconds. Error alerts should be persistent, and close only when action is resolved.

### Types

#### Information

- Used when there is information or tips that users can benefit from
- System-wide or positioned contextually above the information or form they relate to
- Can link to other pages
- Can be closed with a **Close/Dismiss button**
- Can close automatically after 10 seconds

#### Success

- Used when an action has been completed successfully
- Positioned contextually above the information or form they relate to
- Can link to other pages
- Can be closed with a **Close/Dismiss button**
- Can close automatically after 10 seconds

#### Warning

- Used when an action or item needs attention
- Positioned contextually above the information or form they relate to
- Can link to other pages
- Can be closed with a **Close/Dismiss button**
- Should **not** close automatically, unless the action has been resolved

#### Error

- Used when the system has failed to complete an action, or the user has made an error
- Used when an action or item needs attention
- Positioned contextually above the information or form they relate to
- Can link to other pages
- Can be closed with a **Close/Dismiss button**
- Should **not** close automatically, unless the action has been resolved

### Theme

Assume that any theme prop that does not begin with "rui" is within `rui_components`. See [Theming Components](./#!/Theming%20Components).

| Theme Prop | Default | Description |
