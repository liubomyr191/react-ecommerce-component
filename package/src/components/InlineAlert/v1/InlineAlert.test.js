import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import InlineAlert from "./InlineAlert";

test("basic snapshot", () => {
  const component = renderer.create(<InlineAlert />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
