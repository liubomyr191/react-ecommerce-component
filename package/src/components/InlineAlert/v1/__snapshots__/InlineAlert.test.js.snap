// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic error snapshot 1`] = `
.c0 {
  -webkit-font-smoothing: antialiased;
  color: #3c3c3c;
  font-family: 'Source Sans Pro','Helvetica Neue',Helvetica,sans-serif;
  font-size: 16px;
  font-style: normal;
  font-stretch: normal;
  font-weight: 400;
  -webkit-letter-spacing: .03em;
  -moz-letter-spacing: .03em;
  -ms-letter-spacing: .03em;
  letter-spacing: .03em;
  line-height: 1.5;
  border-radius: 2px;
  border-style: solid;
  border-width: 2px;
  padding-bottom: 15px;
  padding-left: 15px;
  padding-right: 15px;
  padding-top: 15px;
  white-space: pre-wrap;
  color: #3c1f21;
  background-color: #ffeeef;
  border-color: #cd3f4c;
}

.c1 {
  -webkit-font-smoothing: antialiased;
  color: #3c3c3c;
  font-family: 'Source Sans Pro','Helvetica Neue',Helvetica,sans-serif;
  font-size: 16px;
  font-style: normal;
  font-stretch: normal;
  font-weight: 600;
  -webkit-letter-spacing: .03em;
  -moz-letter-spacing: .03em;
  -ms-letter-spacing: .03em;
  letter-spacing: .03em;
  line-height: 1.5;
  padding-bottom: 10px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    Error
  </div>
  Card ending in 0000 is expiring soon.
</div>
`;

exports[`basic information snapshot 1`] = `
.c0 {
  -webkit-font-smoothing: antialiased;
  color: #3c3c3c;
  font-family: 'Source Sans Pro','Helvetica Neue',Helvetica,sans-serif;
  font-size: 16px;
  font-style: normal;
  font-stretch: normal;
  font-weight: 400;
  -webkit-letter-spacing: .03em;
  -moz-letter-spacing: .03em;
  -ms-letter-spacing: .03em;
  letter-spacing: .03em;
  line-height: 1.5;
  border-radius: 2px;
  border-style: solid;
  border-width: 2px;
  padding-bottom: 15px;
  padding-left: 15px;
  padding-right: 15px;
  padding-top: 15px;
  white-space: pre-wrap;
  color: #285268;
  background-color: #ecf8fe;
  border-color: #1999dd;
}

.c1 {
  -webkit-font-smoothing: antialiased;
  color: #3c3c3c;
  font-family: 'Source Sans Pro','Helvetica Neue',Helvetica,sans-serif;
  font-size: 16px;
  font-style: normal;
  font-stretch: normal;
  font-weight: 600;
  -webkit-letter-spacing: .03em;
  -moz-letter-spacing: .03em;
  -ms-letter-spacing: .03em;
  letter-spacing: .03em;
  line-height: 1.5;
  padding-bottom: 10px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    Information
  </div>
  Card ending in 0000 is expiring soon.
</div>
`;

exports[`basic snapshot without title 1`] = `
.c0 {
  -webkit-font-smoothing: antialiased;
  color: #3c3c3c;
  font-family: 'Source Sans Pro','Helvetica Neue',Helvetica,sans-serif;
  font-size: 16px;
  font-style: normal;
  font-stretch: normal;
  font-weight: 400;
  -webkit-letter-spacing: .03em;
  -moz-letter-spacing: .03em;
  -ms-letter-spacing: .03em;
  letter-spacing: .03em;
  line-height: 1.5;
  border-radius: 2px;
  border-style: solid;
  border-width: 2px;
  padding-bottom: 15px;
  padding-left: 15px;
  padding-right: 15px;
  padding-top: 15px;
  white-space: pre-wrap;
  color: #7a6322;
  background-color: #fcf3dc;
  border-color: #a2832d;
}

<div
  className="c0"
>
  Card ending in 0000 is expiring soon.
</div>
`;

exports[`basic success snapshot 1`] = `
.c0 {
  -webkit-font-smoothing: antialiased;
  color: #3c3c3c;
  font-family: 'Source Sans Pro','Helvetica Neue',Helvetica,sans-serif;
  font-size: 16px;
  font-style: normal;
  font-stretch: normal;
  font-weight: 400;
  -webkit-letter-spacing: .03em;
  -moz-letter-spacing: .03em;
  -ms-letter-spacing: .03em;
  letter-spacing: .03em;
  line-height: 1.5;
  border-radius: 2px;
  border-style: solid;
  border-width: 2px;
  padding-bottom: 15px;
  padding-left: 15px;
  padding-right: 15px;
  padding-top: 15px;
  white-space: pre-wrap;
  color: #1e4035;
  background-color: #dcfaf1;
  border-color: #158562;
}

.c1 {
  -webkit-font-smoothing: antialiased;
  color: #3c3c3c;
  font-family: 'Source Sans Pro','Helvetica Neue',Helvetica,sans-serif;
  font-size: 16px;
  font-style: normal;
  font-stretch: normal;
  font-weight: 600;
  -webkit-letter-spacing: .03em;
  -moz-letter-spacing: .03em;
  -ms-letter-spacing: .03em;
  letter-spacing: .03em;
  line-height: 1.5;
  padding-bottom: 10px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    Success
  </div>
  Card ending in 0000 is expiring soon.
</div>
`;

exports[`basic warning snapshot 1`] = `
.c0 {
  -webkit-font-smoothing: antialiased;
  color: #3c3c3c;
  font-family: 'Source Sans Pro','Helvetica Neue',Helvetica,sans-serif;
  font-size: 16px;
  font-style: normal;
  font-stretch: normal;
  font-weight: 400;
  -webkit-letter-spacing: .03em;
  -moz-letter-spacing: .03em;
  -ms-letter-spacing: .03em;
  letter-spacing: .03em;
  line-height: 1.5;
  border-radius: 2px;
  border-style: solid;
  border-width: 2px;
  padding-bottom: 15px;
  padding-left: 15px;
  padding-right: 15px;
  padding-top: 15px;
  white-space: pre-wrap;
  color: #7a6322;
  background-color: #fcf3dc;
  border-color: #a2832d;
}

.c1 {
  -webkit-font-smoothing: antialiased;
  color: #3c3c3c;
  font-family: 'Source Sans Pro','Helvetica Neue',Helvetica,sans-serif;
  font-size: 16px;
  font-style: normal;
  font-stretch: normal;
  font-weight: 600;
  -webkit-letter-spacing: .03em;
  -moz-letter-spacing: .03em;
  -ms-letter-spacing: .03em;
  letter-spacing: .03em;
  line-height: 1.5;
  padding-bottom: 10px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    Warning
  </div>
  Card ending in 0000 is expiring soon.
</div>
`;
